src/Data/Mapper/PositionMapper.php

```diff
@@ -27,4 +27,24 @@ public function getPositions($options = [])
 
         return [$this->mapObjects($result), $paginate];
     }
+
+    protected function mapObjects(array $records)
+    {
+        $results = [];
+        foreach ($records as $record) {
+            $entity = new $this->entity;
```

Comment:
I don't particularly think this is the right abstraction of factory pattern. Normally you would want to have:

```php
abstract class DataMapper
{
    public function __construct(EntityFactory $factory)
    {
        $this->factory = $factory;
    }

    public function map(array $results)
    {
        $output = [];
        foreach ($results as $row) {
            $output[] = $this->factory->make($row); // creates Entity
        }
        return $output;
    }
}
```

And then each of mappers:

```php
use Factory/Entity/PositionFactory; // injected factory
class PositionMapper extends DataMapper
{
    public function all()
    {
        // query and ...
        return $this->map($results);
    }
}
```